### ISSUE TITLE
False positive `unexpected-special-method-signature` errors for the __init_subclass__ method

### ISSUE THREAD

**Issue by yilei** [PyCQA/pylint#6644]
### Bug description

Example taken from https://docs.python.org/3/reference/datamodel.html#object.__init_subclass__ (minus the positional only arg):

```
# pylint: disable=too-few-public-methods,missing-class-docstring,missing-module-docstring


class Philosopher:
    def __init_subclass__(cls, default_name, **kwargs):
        super().__init_subclass__(**kwargs)
        cls.default_name = default_name

class AustralianPhilosopher(Philosopher, default_name="Bruce"):
    pass
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:5:4: E0302: The special method '__init_subclass__' expects 0 param(s), 1 was given (unexpected-special-method-signature)
```

### Expected behavior

This should not raise the error as `__init_subclass__` can have any number of parameters.

### Pylint version

```shell
pylint 2.13.9
astroid 2.11.5
Python 3.9.10 (main, Mar 16 2022, 06:54:08)
[GCC 11.2.0]
```

**Comment by mbyrnepr2** [comment]
Looks like a bug to me. Examples of `__init_subclass__` usage from the P[ython documentation](https://docs.python.org/3/reference/datamodel.html#object.__init_subclass__) linked and from the [PEP](https://peps.python.org/pep-0487/) show more than zero arguments used.
Moving [this line](https://github.com/PyCQA/pylint/blob/main/pylint/checkers/utils.py#L110)  to [this line ](https://github.com/PyCQA/pylint/blob/main/pylint/checkers/utils.py#L72) seems to solve it.

**Comment by Pierre-Sassoulas** [comment]
Do you want to make the fix yourself @mbyrnepr2 ?

**Comment by mbyrnepr2** [comment]
> Do you want to make the fix yourself @mbyrnepr2 ?

I've linked a Mr above @Pierre-Sassoulas. But now that you mention it, perhaps that wasn't good etiquette from me; if @yilei wishes to submit a MR then I'm happy to close mine.

**Comment by yilei** [comment]
@mbyrnepr2 IMO this is very welcomed! Thank you for your quick fix, please go ahead with the PR 👍
